### PR TITLE
Remove temp file for logs blob upload

### PR DESCRIPTION
**Pull request by hallipr**
Currently, we're receiving IOExceptions for disk space.

### CODE REVIEW

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs [hunk 1]
```diff
@@ -310,52 +310,48 @@ private async Task UploadLogLinesBlobAsync(string account, Build build, BuildLog
...
 
-                await using (var messagesWriter = new StreamWriter(File.OpenWrite(tempFile.Path)))
+                for (var lineNumber = 1; lineNumber <= logLines.Count; lineNumber++)
                 {
-                    var logLines = await this.logProvider.GetLogLinesAsync(build, log.Id);
-                    var lastTimeStamp = log.CreatedOn;
+                    var line = logLines[lineNumber - 1];
+                    var match = Regex.Match(line, @"^([^Z]{20,28}Z) (.*)$");
+                    var timestamp = match.Success
+                        ? DateTime.ParseExact(match.Groups[1].Value, TimeFormat, null,
```
Comment (scbedd) [inline tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs:322]: Safe assumption due to data format? Maybe drop an example of what that format is actually parsing in a comment so we can specifically tell.

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs [hunk 2]
```diff
@@ -310,52 +310,48 @@ private async Task UploadLogLinesBlobAsync(string account, Build build, BuildLog
...
 
-                    for (var lineNumber = 1; lineNumber <= logLines.Count; lineNumber++)
-                    {
-                        var line = logLines[lineNumber - 1];
-                        var match = Regex.Match(line, @"^([^Z]{20,28}Z) (.*)$");
-                        var timestamp = match.Success
-                            ? DateTime.ParseExact(match.Groups[1].Value, TimeFormat, null,
-                                System.Globalization.DateTimeStyles.AssumeUniversal).ToUniversalTime()
-                            : lastTimeStamp;
+                    var message = match.Success ? match.Groups[2].Value : line;
```
Comment (scbedd) [inline tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs:326]: Same concern here. You're doing this so obviously it must always be like this.

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs [hunk 3]
```diff
@@ -310,52 +310,48 @@ private async Task UploadLogLinesBlobAsync(string account, Build build, BuildLog
...
 
-                        var message = match.Success ? match.Groups[2].Value : line;
+                    if (timestamp == null)
+                    {
+                        throw new Exception($"Error processing line {lineNumber}. No leading timestamp.");
```
Comment (scbedd) [inline tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs:330]: Suggestion: dump the line causing the error too for easy debugging in an unexpected case later.

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs [hunk 4]
```diff
@@ -310,52 +310,48 @@ private async Task UploadLogLinesBlobAsync(string account, Build build, BuildLog
...
+                var messagesWriter = new StringBuilder();
+                var logLines = await this.logProvider.GetLogLinesAsync(build, log.Id);
+                var lastTimeStamp = log.CreatedOn;
 
-                await using (var messagesWriter = new StreamWriter(File.OpenWrite(tempFile.Path)))
+                for (var lineNumber = 1; lineNumber <= logLines.Count; lineNumber++)
                 {
-                    var logLines = await this.logProvider.GetLogLinesAsync(build, log.Id);
-                    var lastTimeStamp = log.CreatedOn;
+                    var line = logLines[lineNumber - 1];
```
Comment (scbedd) [inline tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs:319]: Nit: Why start at 1?

**Review by scbedd — APPROVED** [review]
Questions, but no major concerns.

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs (hunk 1 above)
Comment (hallipr) [reply tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs:322]: good point

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs (hunk 4 above)
Comment (hallipr) [reply tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs:319]: Azure DevOps numbers the lines starting with 1:
image

**Pull request merged**